### PR TITLE
WIP: fixing no-connection issues.

### PR DESCRIPTION
**Pull request by NikolaBorisov**

### CODE REVIEW

**Review by balusu-murali — APPROVED** [review]

File: src/ecredis.erl [hunk 1]
```diff
@@ -337,6 +337,20 @@ get_successes_and_retries(#query{response = {error, <<"TRYAGAIN">>}, retries = R
     % and the keys are currently not in the same node.
     ecredis_logger:log_error("TRYAGAIN error", Query),
     {[], [Query#query{retries = Retries + 1}]};
+get_successes_and_retries(#query{response = {error, no_connection},
```
Comment (vipin211) [inline src/ecredis.erl:340]: Remove the TODO on line 358.

**Review by vipin211 — APPROVED** [review]

File: src/ecredis.erl [hunk 2]
```diff
@@ -337,6 +337,20 @@ get_successes_and_retries(#query{response = {error, <<"TRYAGAIN">>}, retries = R
...
+                [ClusterName, OldPid, Pid, Query]),
+            {[], [Query#query{retries = Retries + 1, pid = Pid, version = NewVersion}]};
+        undefined ->
+            ?ERROR("Error no_connection ~p no pid for query after remap ~p~n",
+                [ClusterName, Query]),
+            {[], [Query#query{retries = Retries + 1}]}
+    end;
 get_successes_and_retries(#query{response = {error, _}, retries = Retries} = Query) ->
     % TODO fill in handlers for other errors, as for when to retry or when to not
     % - CLUSTERDOWN should retry
```
Comment (balusu-murali) [inline src/ecredis.erl:356]: this is also a possible error - noticed it for redis-whisper here.
https://sentry.io/organizations/halloapp/issues/2211537526/?project=5507654&query=is%3Aunresolved

i think we should add this remap_cluster even for this error as well.
cc: @vipin211 @NikolaBorisov 
what do you think?

Comment (NikolaBorisov) [reply src/ecredis.erl:356]: #60

File: src/ecredis.erl (hunk 1 above)
Comment (NikolaBorisov) [reply src/ecredis.erl:340]: fixed in #60

**Pull request merged**